### PR TITLE
[improve](partition) support auto list partition with more columns

### PR DESCRIPTION
**Pull request by zhangstar333**
## Proposed changes
before the partition by column only have one column.
now remove those limit, could have more columns.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/sink/vrow_distribution.cpp [hunk 1]
```diff
@@ -29,28 +30,32 @@
...
+Status VRowDistribution::_save_missing_values(const std::vector<std::vector<std::string>>& col_strs,
+                                              int col_size, Block* block,
                                               std::vector<int64_t> filter) {
     // de-duplication for new partitions but save all rows.
     _batching_block->add_rows(block, filter);
-    for (auto row : filter) {
-        auto val_str = value_type->to_string(*col, row);
-        if (!_deduper.contains(val_str)) {
-            _deduper.emplace(val_str);
+    _partitions_need_create.clear();
```
Comment (zclllyybb) [inline be/src/vec/sink/vrow_distribution.cpp:48]: we use batching here. don't clear it.

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run p0 pipelineX

File: be/src/vec/sink/vrow_distribution.cpp [hunk 2]
```diff
@@ -29,28 +30,31 @@
...
-        if (!_deduper.contains(val_str)) {
-            _deduper.emplace(val_str);
+    std::vector<TStringLiteral> cur_row_values;
+    for (int row = 0; row < col_strs[0].size(); ++row) {
+        cur_row_values.clear();
+        for (int col = 0; col < col_size; ++col) {
             TStringLiteral node;
-            node.value = std::move(val_str);
-            _partitions_need_create.emplace_back(std::vector {node}); // only 1 partition column now
+            node.value = col_strs[col][row];
```
Comment (HappenLee) [inline be/src/vec/sink/vrow_distribution.cpp:53]: why not use `std::move` ?

**Comment by zhangstar333** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

File: regression-test/suites/partition_p0/auto_partition/test_auto_list_partition.groovy [hunk 3]
```diff
@@ -230,4 +230,44 @@ suite("test_auto_list_partition") {
...
+        )
+        AUTO PARTITION BY LIST (`id`, `k`, `str`)
+        (
+        )
+        DISTRIBUTED BY HASH(`k`) BUCKETS 16
+        PROPERTIES (
+            "replication_allocation" = "tag.location.default: 1"
+        );
+    """
+    sql """ insert into test_list_many_column2 values (1,1,"asd"), (-1,-1,"vdf");"""
```
Comment (zclllyybb) [inline regression-test/suites/partition_p0/auto_partition/test_auto_list_partition.groovy:268]: maybe we should add some cases which have more chance to conflict? like '! ' and '!'; '20' and ' '; and so on

**Review by zclllyybb — APPROVED** [review]

**Pull request merged**